### PR TITLE
[SPARK-20997][CORE]'--driver-cores'  standalone or Mesos or YARN in Cluster deploy mode only.

### PR DESCRIPTION
**Pull request by guoxiaolongzte**
## What changes were proposed in this pull request?

'--driver-cores'  standalone or Mesos or YARN in Cluster deploy mode only.So  The description of spark-submit about it is not very accurate.

## How was this patch tested?

manual tests

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala [hunk 1]
```diff
@@ -558,8 +558,9 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
         |  --verbose, -v               Print additional debug output.
         |  --version,                  Print the version of current Spark.
         |
-        | Spark standalone with cluster deploy mode only:
-        |  --driver-cores NUM          Cores for driver (Default: 1).
+        | Cluster deploy mode only:
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala]: This is probably fine, but, does it apply to Mesos cluster mode?

Comment (guoxiaolongzte) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala]: Yes, I see the code used, although I am not familiar with mesos.
image

Comment (guoxiaolongzte) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala]: Is it ok to modify this?

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**